assignment2/q2.cpp

```diff
@@ -0,0 +1,148 @@
+#include <iostream>
+using namespace std;
+
+struct Node{
+    int value;
+    Node *parent;
+    Node *left, *right;
+};
+
+bool FindNode(Node *root, Node *to_find) {
```

Comment:
The name `FindNode` (similarly to FindKey) indicates that you look for something in the tree and return it.
A better name for something that returns `bool` is `NodeExists` or `DoesNodeExistInTree`.
Or even `IsAncestorOf` (because `root` is ancestor of `to_find`).